release/ray_release/test_automation/state_machine.py

```diff
@@ -21,48 +29,114 @@ class TestStateMachine:
     """
 
     ray_repo = None
+    ray_buildkite = None
 
     def __init__(self, test: Test) -> None:
         self.test = test
         self.test_results = test.get_test_results()
-        self.state_machine = {
-            # jailed -> passing
-            TestState.JAILED: TestState.PASSING
-            if self._jailed_to_passing()
-            else TestState.JAILED,
-            # passing -> failing
-            TestState.PASSING: TestState.FAILING
-            if self._passing_to_failing()
-            else TestState.PASSING,
-            # failing -> passing OR failing -> jailed
-            TestState.FAILING: TestState.PASSING
-            if self._failing_to_passing()
-            else (TestState.JAILED if self._failing_to_jailed() else TestState.FAILING),
-        }
         if not self.ray_repo:
-            github_token = TestStateMachine.github_token = boto3.client(
-                "secretsmanager", region_name="us-west-2"
-            ).get_secret_value(SecretId=AWS_SECRET_GITHUB)["SecretString"]
+            github_token = get_secret_token(AWS_SECRET_GITHUB)
             self.ray_repo = Github(github_token).get_repo(RAY_REPO)
+        if not self.ray_buildkite:
+            buildkite_token = get_secret_token(AWS_SECRET_BUILDKITE)
+            self.ray_buildkite = Buildkite()
+            self.ray_buildkite.set_access_token(buildkite_token)
 
     def move(self) -> None:
         """
         Move the test to the next state.
         """
         from_state = self.test.get_state()
-        to_state = self.state_machine[from_state]
+        to_state = self._next_state(from_state)
         self.test.set_state(to_state)
         self._move_hook(from_state, to_state)
 
+    def _next_state(self, current_state) -> TestState:
+        """
+        Compute the next state of the test based on the current state and the test
+        """
+        if current_state == TestState.PASSING:
+            if self._passing_to_consistently_failing():
+                return TestState.CONSITENTLY_FAILING
+            if self._passing_to_failing():
+                return TestState.FAILING
```

Comment:
so, in other cases, it will be kept in PASSING state?

Reply:
Other cases they would remain at the same state yes